### PR TITLE
(#826) Envelopes for Scalar and Func

### PR DESCRIPTION
**Pull request by victornoel**
For #826.

Nothing special here, I added some very basic tests too.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1091](https://github.com/yegor256/cactoos/issues/1091) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1091](https://github.com/yegor256/cactoos/issues/1091) is assigned to @atapin[/z](https://www.0crat.com/u/atapin), here is [why](https://www.0crat.com/footprint/C63314D6Z/af0e1773-1154-4809-8d27-fe18d48ed41c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/org/cactoos/func/FuncEnvelope.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * <p>There is no thread-safety guarantee.
+ *
+ * @param <X> Type of input
+ * @param <Y> Type of output
+ * @since 0.41
+ */
+public abstract class FuncEnvelope<X, Y> implements Func<X, Y> {
+
+    /**
+     * The function.
```
Comment (atapin) [inline src/main/java/org/cactoos/func/FuncEnvelope.java]: The only thing I would fix is the javadoc – ~The function~ The delegate function

File: src/main/java/org/cactoos/scalar/ScalarEnvelope.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @param <T> Type of result
+ * @since 0.41
+ */
+public abstract class ScalarEnvelope<T> implements Scalar<T> {
+
+    /**
+     * The scalar.
```
Comment (atapin) [inline src/main/java/org/cactoos/scalar/ScalarEnvelope.java]: ~The scalar~ The delegate scalar

**Comment by victornoel** [comment]
@atapin thanks, I've pushed the changes

**Comment by atapin** [comment]
@llorllale the PR looks good

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 22 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#1091](https://github.com/yegor256/cactoos/issues/1091) is not in the agenda of @atapin[/z](https://www.0crat.com/u/atapin), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @atapin[/z](https://www.0crat.com/u/atapin), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1091](https://github.com/yegor256/cactoos/issues/1091) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @atapin[/z](https://www.0crat.com/u/atapin)

**Pull request merged**